test_crates/function_parameter_type_changed/new/src/lib.rs

```diff
@@ -0,0 +1,9 @@
+#![no_std]
+
+// This should trigger the lint using witnesses
+pub fn function_with_parameter_changed(_: i64) {}
+
+// This should be detected by the lint, but should be ignored since it won't trigger the witness
+pub fn function_with_signature_changed(_param_changed: ()) {}
```

Comment:
Other good cases in this vein:
```rust
// old
pub fn explicit_arg_to_impl_trait(_: i64) {}
// new
pub fn explicit_arg_to_impl_trait(_: impl Into<i64>) {}

// old
pub fn impl_trait_to_generic(_: impl Into<i64>) {}
// new
pub fn impl_trait_to_generic<T: Into<i64>>(_: T) {}

// old
pub fn impl_trait_to_generic_where_bound(_: impl Into<i64>) {}
// new
pub fn impl_trait_to_generic_where_bound<T>(_: T) where T: Into<i64> {}
```

The last case is symmetric, neither direction is a breaking change. The others are not breaking in the old-to-new direction, but they are breaking from new-to-old---more good cases to lint.